docs(pie-docs): DSW-2220 add overview page and assets (#1605)
justeattakeaway/pie: apps/pie-docs/src/components/checkbox/overview/overview.md

---
eleventyNavigation:
    key: Overview
    parent: 'Checkbox'
    order: 1
shouldShowContents: true
permalink: components/checkbox/
---

## Overview

Checkboxes provide a binary selection mechanism where users can either check the box to indicate they want a particular option or leave it unchecked to indicate they do not want that option.

They are commonly used in forms, settings panels, and various other interfaces where users need to make a single selection.

{% contentPageImage {
    src:"../../../assets/img/components/checkbox/overview.svg",
    alt: "Checkbox labeled 'I have read and accepted the terms and conditions' is checked."
} %}

---

## Dos and Don’ts

{% usage {
    do: {
        type: usageTypes.text,
        items: [

>           "Ensure labels are concise and descriptive that clearly indicates what selecting it will entail."

Reviewer: Yeah that's incorrect, also unnecessarily complex wording 😄 
I think this is what it's trying to say:
```suggestion
          "Make sure labels are concise and clearly describe what checking the box means."
```

### Label

A label should always be present unless being utilised as a nested component and an alternative label is present.

{% contentLayout %}
  {% contentItem %}
    {% contentPageImage {
      src: "../../../assets/img/components/checkbox/checkbox-default.svg",
      width: "70px",
      alt: "A blank checkbox followed by the label text."
    } %}
  {% endcontentItem %}
  {% contentItem %}
    {% contentPageImage {
      src: "../../../assets/img/components/checkbox/checkbox.svg",
      width: "24px",
      alt: "A blank checkbox component."
    } %}
  {% endcontentItem %}
{% endcontentLayout %}

### Assistive text

Assistive text is used where needed to provide additional information or error / success messaging. The assistive text is always used for an error state to provide the user with the information required to complete the form element.

{% contentPageImage {
    src:"../../../assets/img/components/checkbox/checkbox-with-assistive-text.svg",
    alt: "A blank checkbox followed by assistive text in red below it.",
    width: "112px"
} %}

---

## Content

All content should use sentence case.

### Label

- Always use clear and concise labels for checkboxes.

> - Labels appear to the right of checkbox input.

Reviewer: Thoughts?
```suggestion
- Labels appear to the right of checkbox input (when the text direction is left-to-right).
```

      width: "70px",
      alt: "Checkbox component when hovered over."
    } %}
  {% endcontentItem %}
  {% contentItem %}
    <h4>Active</h4>
    {% contentPageImage {
      src: "../../../assets/img/components/checkbox/checkbox-active.svg",
      width: "70px",
      alt: "Checkbox component in an active state."
    } %}
  {% endcontentItem %}
  {% contentItem %}
    <h4>Focus</h4>
    {% contentPageImage {
      src: "../../../assets/img/components/checkbox/checkbox-focused.svg",
      width: "70px",
      alt: "Checkbox component when it has focus."
    } %}
  {% endcontentItem %}
  {% contentItem %}
    <h4>Disabled</h4>
    {% contentPageImage {
      src: "../../../assets/img/components/checkbox/checkbox-disabled.svg",
      width: "70px",
      alt: "Checkbox component in a disabled state."
    } %}
  {% endcontentItem %}
  {% contentItem %}
    <h4>Error</h4>
    {% contentPageImage {
      src: "../../../assets/img/components/checkbox/checkbox-error.svg",
      width: "112px",
      alt: "Checkbox state example showing an error."
    } %}
  {% endcontentItem %}
{% endcontentLayout %}

---

## Examples

> Outlines the atomic level interactive elements for the component.
> 

Reviewer: I think I remember there was a discussion saying that this piece of text only belongs in the "Interactive states" section above. @tingting-lau?

```suggestion
```

Reviewer 2: That is correct :) Thanks Xander.

### LTR examples

> Here are some examples of the component in left-to-right context:

Reviewer: ```suggestion
Here are some examples of the component in a left-to-right context:
```

{% contentLayout %}
  {% contentItem %}
    {% contentPageImage {
      src: "../../../assets/img/components/checkbox/checkbox-LTR-agreement.svg",
      width: "375px",
      alt: "Checkbox labeled 'I have read and accepted the terms and conditions' in left-to-right context."
    } %}
  {% endcontentItem %}
  {% contentItem %}
    {% contentPageImage {
      src: "../../../assets/img/components/checkbox/checkbox-checked-LTR.svg",
      width: "360px",

>       alt: "Selected checkbox followed by the label text in left-to-right context."

Reviewer: ```suggestion
      alt: "Selected checkbox followed by the label text in a left-to-right context."
```

    } %}
  {% endcontentItem %}
{% endcontentLayout %}

### RTL examples

Here are some examples of the component in right-to-left context:

{% contentLayout %}
  {% contentItem %}
    {% contentPageImage {
      src: "../../../assets/img/components/checkbox/checkbox-RTL-agreement.svg",
      width: "271px",
      alt: "Checkbox labeled 'I have read and accepted the terms and conditions' in right-to-left context."
    } %}
  {% endcontentItem %}
  {% contentItem %}
    {% contentPageImage {
      src: "../../../assets/img/components/checkbox/checkbox-checked-RTL.svg",
      width: "360px",

>       alt: "Selected checkbox followed by the label text in right-to-left context."

Reviewer: ```suggestion
      alt: "Selected checkbox followed by the label text in a right-to-left context."
```
